docs/concepts/bitswap.md

```diff
@@ -11,4 +10,54 @@ related:
 
 # Bitswap
 
-<ContentStatus />
+Bitswap is a core module of IPFS for exchanging blocks of data. It directs the requesting and sending of blocks to and from other peers in the network. Bitswap is a _message-based protocol_ where all messages contain [wantlists](#wantlist) or blocks. Bitswap has a [Go implementation](https://github.com/ipfs/go-bitswap) and a [JavaScript implementation](https://github.com/ipfs/js-ipfs-bitswap).
+
+Bitswap has two main jobs:
+- to acquire blocks requested by the client from the network.
+- to send blocks in its possession to other peers who want them.
+
+### Wantlist
+
+A wantlist is a data structure that an IPFS node sends to the network to tell peers which blocks it wants. The wantlist contains [CIDs](https://docs-beta.ipfs.io/concepts/content-addressing/) (content-addressed identifiers that refer to particular blocks desired) along with priority and cancellation information.
+
+### Message
+
+A single Bitswap message may contain any of the following content:
+
+-  **The sender’s wantlist.** This wantlist may either be the sender’s complete wantlist or just the changes to the sender’s wantlist that the receiver needs to know.
+-  **Data blocks.** These are blocks that have been requested (i.e., blocks in the sender’s wantlist, as far as the receiver is aware at the time of message receipt).
+
+### Session
+
+IPFS content blocks are often connected to each other through a [Merkle DAG](https://docs-beta.ipfs.io/concepts/merkle-dag/). If a sender's block requests are related, a mechanism called a _Bitswap session_ can optimize block requests sent to other peers. This mechanism can increase transfer speed and reduce the number of duplicate blocks on the network. Active peers are favored, as they are tracked relative to them having the requested blocks and how quickly they respond.
```

Comment:
The gist of a bitswap session is that if a remote peer(s) has one block, it is likely to have other blocks from the same DAG, so new blocks are preferentially requested from that/those peers (thus skipping more expensive paths to content discovery).

This is what this says, but language here is a bit convoluted and it took me a couple of reads.